packages/block-editor/src/components/rich-text/index.js

```diff
@@ -2,1112 +2,128 @@
-						</>
+const wrapperClasses = 'editor-rich-text block-editor-rich-text';
+const classes = 'editor-rich-text__editable block-editor-rich-text__editable';
+
+function RichTextWraper( {
+	tagName,
+	value: originalValue,
+	onChange: originalOnChange,
+	selectionStart,
+	selectionEnd,
+	onSelectionChange,
+	multiline,
+	onTagNameChange,
+	inlineToolbar,
+	wrapperClassName,
+	className,
+	autocompleters,
+	onReplace,
+	onRemove,
+	onMerge,
+	onSplit,
+	isCaretWithinFormattedText,
+	onEnterFormattedText,
+	onExitFormattedText,
+	canUserUseUnfilteredHTML,
+	isSelected: originalIsSelected,
+	onCreateUndoLevel,
+	placeholder,
+	keepPlaceholderOnFocus,
+	// From experimental filter.
+	...experimentalProps
+} ) {
+	let adjustedValue = originalValue;
+	let adjustedOnChange = originalOnChange;
+
+	// Handle deprecated format.
+	if ( Array.isArray( originalValue ) ) {
+		adjustedValue = children.toHTML( originalValue );
+		adjustedOnChange = ( newValue ) => originalOnChange( children.fromDOM(
+			__unstableCreateElement( document, newValue ).childNodes
+		) );
+	}
+
+	return (
+		<RichText
+			{ ...experimentalProps }
+			value={ adjustedValue }
+			onChange={ adjustedOnChange }
+			selectionStart={ selectionStart }
+			selectionEnd={ selectionEnd }
+			onSelectionChange={ onSelectionChange }
+			tagName={ tagName }
+			wrapperClassName={ classnames( wrapperClasses, wrapperClassName ) }
+			className={ classnames( classes, className ) }
+			placeholder={ placeholder }
+			keepPlaceholderOnFocus={ keepPlaceholderOnFocus }
+			__unstableIsSelected={ originalIsSelected }
+			__unstablePatterns={ getPatterns() }
+			__unstableEnterPatterns={ getEnterPatterns() }
+			__unstablePasteHandler={ pasteHandler }
+			__unstableAutocomplete={ Autocomplete }
+			__unstableAutocompleters={ autocompleters }
+			__unstableOnReplace={ onReplace }
+			__unstableOnRemove={ onRemove }
+			__unstableOnMerge={ onMerge }
+			__unstableOnSplit={ onSplit }
+			__unstableMultiline={ multiline }
+			__unstableIsCaretWithinFormattedText={ isCaretWithinFormattedText }
+			__unstableOnEnterFormattedText={ onEnterFormattedText }
+			__unstableOnExitFormattedText={ onExitFormattedText }
+			__unstableCanUserUseUnfilteredHTML={ canUserUseUnfilteredHTML }
+			__unstableOnCreateUndoLevel={ onCreateUndoLevel }
+		>
+			{ ( { isSelected, value, onChange } ) =>
+				<>
+					{ isSelected && multiline === 'li' && (
+						<ListEdit
+							onTagNameChange={ onTagNameChange }
+							tagName={ tagName }
+							value={ value }
+							onChange={ onChange }
+						/>
 					) }
-				</Autocomplete>
-				{ isSelected && <RemoveBrowserShortcuts /> }
-			</div>
-		);
-	}
+					{ isSelected && ! inlineToolbar && (
+						<BlockFormatControls>
+							<FormatToolbar />
+						</BlockFormatControls>
+					) }
+					{ inlineToolbar && (
+						<IsolatedEventContainer>
+							<FormatToolbar />
+						</IsolatedEventContainer>
+					) }
+					{ isSelected && <RemoveBrowserShortcuts /> }
```

Comment:
Why aren't these absorb by the underlying component (toolbars and RemoveBrowserShortcuts)?

Reply:
It's block editor specific?

Reply:
I'm not sure what RemoveBrowserShortcuts is but don't we want the format toolbar support by default in the RichText component?

Reply:
It has toolbar support, but not a specific toolbar. This is the toolbar implementation for RichText instances in blocks. When using RichText (without the context), you could create your own custom toolbars.